src/Compilers/Core/Portable/InternalUtilities/ReadOnlyHashSet.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Roslyn.Utilities
+{
+    internal struct ReadOnlyHashSet<T> : IReadOnlySet<T>
```

Comment:
Looks like SpecializedCollections.ReadOnlySet(IEnumerable) is not used, only SpecializedCollections.ReadOnlySet(ISet) is. How about changing return type of former to IReadOnlySet and rename latter to ReadOnlySetWrapper? Mind you, that's not consistent with the rest of SpecializedCollections.